### ISSUE TITLE
Can't save a trained model as a HuggingFace model

### ISSUE THREAD

**Issue by Mukhammadsaid19** [mosaicml/examples#400]
After the training has been done, I tried to use a utility function `write_huggingface_pretrained_from_composer_checkpoint` to save a HF model from a checkpoint. However, it throws a key error, that it can't find `huggingface` field inside binary model.

Seems like the saved binaries have their `integrations` field empty. I didn't change any configs inside a Trainer. What might go wrong?

```
File /opt/conda/lib/python3.10/site-packages/composer/models/huggingface.py:522, in get_hf_config_from_composer_state_dict(state_dict, config_overrides)
    519 if config_overrides is None:
    520     config_overrides = {}
--> 522 hf_config_dict = state_dict['state']['integrations']['huggingface']['model']['config']['content']
    523 # Update the config with any extra args needed
    524 hf_config_dict.update(config_overrides)

KeyError: 'huggingface'
```

**Comment by dakinggg** [comment]
Can you share your training script/yaml?

**Comment by dakinggg** [comment]
Closing due to inactivity.

**Comment by FinTexIFB** [comment]
If I may add to this:

I have the problem too after pretraining the hf-version of BERT on all 8 GPUs of a DGX A100 node. The problem is that the following dictionary is empty:
`state_dict['state']['integrations']`

Strangly, if I only use 1 GPU of a DGX A100 node, everything is fine. My workaround at the moment si simply to load the `state_dict['state']['integrations']` of the 1GPU run when I transform the 8GPU run, but that is obviously not ideal. 

I append my traning script and yaml, m
[train_script.txt](ht
[german-fin-hf-bert-base-cased.txt](https://github.com/mosaicml/examples/files/12366715/german-fin-hf-bert-base-cased.txt)
tps://github.com/mosaicml/examples/files/12366707/train_script.txt)
aybe it helps.

**Comment by dakinggg** [comment]
I think that is a bug in an older version of composer. It should be fixed as of composer version 0.15

**Comment by FinTexIFB** [comment]
I use version 0.15.1.
